src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -5201,3 +5211,18 @@ void Compiler::getHWIntrinsicImmOps(NamedIntrinsic    intrinsic,
 }
 
 #endif // FEATURE_HW_INTRINSICS
+
+#if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
+GenTree* Compiler::gtNewSIMDDivByZeroCheck(GenTree*     op,
+                                           var_types    type,
+                                           CorInfoType  simdBaseJitType,
+                                           unsigned int simdSize)
+{
+    assert(type == TYP_INT);
+    GenTree* zeroVecCon = gtNewZeroConNode(op->TypeGet());
+    GenTree* cmpOp      = gtNewSimdCmpOpNode(GT_EQ, op->TypeGet(), op, zeroVecCon, simdBaseJitType, simdSize);
+    GenTreeSIMDDivByZeroChk* hwIntrinsicChk =
+        new (this, GT_SIMD_DIV_BY_ZERO_CHECK) GenTreeSIMDDivByZeroChk(cmpOp, SCK_DIV_BY_ZERO);
```

Comment:
Ah, I see. You're inserting the logic for doing the compare here.

This puts a very specific shape requirement on the `GenTreeSimdDivByZeroChk` node for correctness and it might be better in that scenario to just have such a node take a `bool` operand instead. This would allow you to use `gtNewSimdCmpOpAnyNode` for the common path instead. Lowering can then be special cased to consume the output directly so it can use the `bool` or the direct flags, as other cases do with such flags producing nodes.

Reply:
Oh I see, that does seem better. I will work on making the adjustment.